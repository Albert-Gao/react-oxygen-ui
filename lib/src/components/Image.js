"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styles_1 = require("../styles");
const utils_1 = require("../utils/utils");
const BaseBox_1 = require("./Box/BaseBox");
exports.Image = ({ height, css, takeRef, ...rest }) => (<BaseBox_1.BaseBox as="img" maxWidth="100%" height={utils_1.is.notExist(height) ? 'auto' : height} css={`
      object-fit: contain;
      ${styles_1.getSafeCss(css)};
    `} {...utils_1.getRefProp(takeRef)} {...rest}/>);
