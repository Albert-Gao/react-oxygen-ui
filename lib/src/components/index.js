"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./Box/BaseBox"));
__export(require("./Box/Box"));
__export(require("./Buttons/Button"));
__export(require("./Buttons/LoadingButton"));
__export(require("./Buttons/FlatButton"));
__export(require("./Inputs/Input"));
__export(require("./Inputs/Textarea"));
__export(require("./Text/Heading"));
__export(require("./Text/Paragraph"));
__export(require("./Text/Text"));
__export(require("./Image"));
