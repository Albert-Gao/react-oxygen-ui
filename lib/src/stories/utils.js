"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const addon_actions_1 = require("@storybook/addon-actions");
const addon_knobs_1 = require("@storybook/addon-knobs");
const styles_1 = require("../styles");
function arrayToObject(data, keyToMap) {
    if (!data || data.length === 0) {
        return {};
    }
    const result = {};
    data.forEach((item, index) => {
        const i = keyToMap ? keyToMap[index] : index;
        result[i] = item;
    });
    return result;
}
exports.arrayToObject = arrayToObject;
// We wrap the action to prevent been display in addon-jsx
function wrapAction(actionName) {
    const wrapped = addon_actions_1.action(actionName);
    wrapped.toString = () => `action('${actionName}')`;
    return wrapped;
}
exports.wrapAction = wrapAction;
const keyOfColors = Object.keys(styles_1.defaultTheme.colors);
exports.Knobs = {
    spacesKnob: (propName, defaultValue = 2) => addon_knobs_1.select(propName, styles_1.DEFAULT_SPACE, defaultValue),
    colorKnob: (propName, defaultValue = 'primary') => addon_knobs_1.select(propName, arrayToObject(keyOfColors, keyOfColors), defaultValue),
    fontWeightKnob: (defaultValue = 'regular') => addon_knobs_1.select('fontWeight', {
        thin: 'thin',
        light: 'light',
        regular: 'regular',
        medium: 'medium',
        bold: 'bold',
        black: 'black',
    }, defaultValue),
    fontSizeKnob: (propName = 'fontSize', defaultValue = 1) => addon_knobs_1.select(propName, styles_1.DEFAULT_FONT_SIZE, defaultValue),
    sizeKnob: (propName, defaultValue = 'medium') => addon_knobs_1.select(propName, styles_1.DEFAULT_SIZE, styles_1.DEFAULT_SIZE[defaultValue]),
};
