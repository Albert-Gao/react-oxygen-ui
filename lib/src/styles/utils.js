"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled_system_1 = require("styled-system");
const utils_1 = require("../utils/utils");
exports.getMediaQuery = (screenSize, css) => `${styled_system_1.createMediaQuery(screenSize)} {
    ${css};
  }`;
exports.getTransitionValue = (propertyNames = 'transform') => `${propertyNames} 0.3s cubic-bezier(0.53, 0.01, 0.36, 1.63);`;
function getSafeValue(target, defaultValue) {
    return utils_1.is.notExist(target) ? defaultValue : target;
}
exports.getSafeValue = getSafeValue;
function getSafeCss(target) {
    return getSafeValue(target, '');
}
exports.getSafeCss = getSafeCss;
