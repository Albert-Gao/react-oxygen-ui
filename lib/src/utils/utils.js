"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.is = {
    function: (v) => typeof v === 'function',
    number: (v) => typeof v === 'number',
    notExist: (v) => v === undefined || v === null,
};
exports.getRefProp = (takeRef) => takeRef
    ? {
        ref: takeRef,
    }
    : {};
