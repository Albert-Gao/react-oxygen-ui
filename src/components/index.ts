export * from './Box/BaseBox';
export * from './Box/BaseBox.type';

export * from './Box/Box';

export * from './Buttons/Button';
export * from './Buttons/LoadingButton';
export * from './Buttons/FlatButton';

export * from './Inputs/Input';
export * from './Inputs/Textarea';

export * from './Text/Heading';
export * from './Text/Paragraph';
export * from './Text/Text';

export * from './Image';
