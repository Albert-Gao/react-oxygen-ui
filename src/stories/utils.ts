import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { Responsive } from '../components';
import {
  ColorType,
  DEFAULT_FONT_SIZE,
  DEFAULT_SIZE,
  DEFAULT_SPACE,
  defaultTheme,
  FontWeightType,
} from '../styles';

export function arrayToObject<T>(
  data: T[],
  keyToMap?: string[],
): { [id: string]: T } {
  if (!data || data.length === 0) {
    return {};
  }

  const result: { [id: string]: T } = {};
  data.forEach((item, index) => {
    const i = keyToMap ? keyToMap[index] : index;
    result[i] = item;
  });

  return result;
}

// We wrap the action to prevent been display in addon-jsx
export function wrapAction(actionName: string) {
  const wrapped = action(actionName);
  wrapped.toString = () => `action('${actionName}')`;

  return wrapped;
}

const keyOfColors = Object.keys(defaultTheme.colors) as ColorType[];

export const Knobs = {
  spacesKnob: (propName: string, defaultValue: number = 2) =>
    select<number>(propName, DEFAULT_SPACE, defaultValue),

  colorKnob: (
    propName: string,
    defaultValue: ColorType = 'primary',
  ) =>
    select<string>(
      propName,
      arrayToObject<string>(
        keyOfColors as string[],
        keyOfColors as string[],
      ),
      defaultValue,
    ),

  fontWeightKnob: (defaultValue: FontWeightType = 'regular') =>
    select(
      'fontWeight',
      {
        thin: 'thin',
        light: 'light',
        regular: 'regular',
        medium: 'medium',
        bold: 'bold',
        black: 'black',
      },
      defaultValue,
    ),

  fontSizeKnob: (
    propName: string = 'fontSize',
    defaultValue: Responsive = 1,
    // @ts-ignore
  ) => select(propName, DEFAULT_FONT_SIZE, defaultValue),

  sizeKnob: (
    propName: string,
    defaultValue: keyof typeof DEFAULT_SIZE = 'medium',
  ) => select(propName, DEFAULT_SIZE, DEFAULT_SIZE[defaultValue]),
};
