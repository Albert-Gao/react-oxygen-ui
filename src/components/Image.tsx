import * as React from 'react';
import { getSafeCss } from '../styles/utils';
import { getRefProp, is } from '../utils/utils';
import { BaseBox } from './Box/BaseBox';
import { IBaseBoxWithTakeRef } from './Box/BaseBox.type';

export interface IImage extends IBaseBoxWithTakeRef {
  src: string;
  alt?: string;
}

export const Image: React.FC<IImage> = ({ height, css, takeRef, ...rest }) => (
  <BaseBox
    as="img"
    height={is.notExist(height) ? 'auto' : height}
    css={`
      max-width: '100%';
      ${getSafeCss(css)};
    `}
    {...getRefProp(takeRef)}
    {...rest}
  />
);
