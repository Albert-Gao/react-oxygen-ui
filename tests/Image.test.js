import 'jest-styled-components';
import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Image } from '../src/components/Image';
import { TestThemeProvider as ThemeProvider } from './utils';
import { css as cssFunc } from 'styled-components';

afterEach(cleanup);

describe('Test <Image />: ', () => {
  test('css prop should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Image css={'background-color: black;'} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    // the new css is there
    expect(firstChild).toHaveStyleRule('background-color', 'black');

    expect(firstChild.tagName).toEqual('IMG');
  });
});

describe('Test <Image css={}/>: ', () => {
  test('css prop should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Image css={'background-color: black;'} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
  });

  test('css prop (object) should be handled', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Image css={{ backgroundColor: 'black', paddingLeft: '1023px' }} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
    expect(firstChild).toHaveStyleRule('padding-left', '1023px');
  });

  test('css prop (css string) should be handled', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Image
          css={cssFunc`
          background-color: black;
          padding-left: 1023px;
          `}
        />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
    expect(firstChild).toHaveStyleRule('padding-left', '1023px');
  });
});
