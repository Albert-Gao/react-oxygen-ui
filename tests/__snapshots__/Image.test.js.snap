// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Image />:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  height: auto;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  max-width: 100%;
  background-color: black;
}

<img
  class="c0"
  height="auto"
/>
`;

exports[`Test <Image css={}/>:  css prop (css string) should be handled 1`] = `
.c0 {
  box-sizing: border-box;
  height: auto;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  max-width: 100%;
  background-color: black;
  padding-left: 1023px;
}

<img
  class="c0"
  height="auto"
/>
`;

exports[`Test <Image css={}/>:  css prop (object) should be handled 1`] = `
.c0 {
  box-sizing: border-box;
  height: auto;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  max-width: 100%;
  background-color: black;
  padding-left: 1023px;
}

<img
  class="c0"
  height="auto"
/>
`;

exports[`Test <Image css={}/>:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  height: auto;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  max-width: 100%;
  background-color: black;
}

<img
  class="c0"
  height="auto"
/>
`;
